src/MyPlot/MyPlot.php

```diff
@@ -686,6 +688,77 @@ public function clearPlot(Plot $plot, int $maxBlocksPerTick = 256) : bool {
 		return true;
 	}
 
+    /**
+     * Fills the whole plot with a block
+     *
+     * @api
+     *
+     * @param Plot $plot
+     * @param int $plotFillBlock
+     * @param Block $maxBlocksPerTick
+     *
+     * @return bool
+     */
+    public function fillPlot(Plot $plot, Block $plotFillBlock, int $maxBlocksPerTick = 256) : bool {
+        $ev = new MyPlotFillEvent($plot, $maxBlocksPerTick);
+        $ev->call();
+        if($ev->isCancelled()) {
+            return false;
+        }
+        $plot = $ev->getPlot();
+        if(!$this->isLevelLoaded($plot->levelName)) {
+            return false;
+        }
+        $maxBlocksPerTick = $ev->getMaxBlocksPerTick();
+        foreach($this->getServer()->getLevelByName($plot->levelName)->getEntities() as $entity) {
+            if($this->getPlotBB($plot)->isVectorInXZ($entity) && $entity->getY() <= $this->getLevelSettings($plot->levelName)->groundHeight) {
+                if(!$entity instanceof Player) {
+                    $entity->flagForDespawn();
+                }else{
+                    $this->teleportPlayerToPlot($entity, $plot);
+                }
+            }
+        }
+        if($this->getConfig()->get("FastFilling", false)) {
+            $styler = $this->getServer()->getPluginManager()->getPlugin("WorldStyler");
+            if(!$styler instanceof WorldStyler) {
+                return false;
+            }
+            $plotLevel = $this->getLevelSettings($plot->levelName);
+            $plotSize = $plotLevel->plotSize-1;
+            $plotBeginPos = $this->getPlotPosition($plot);
+            $plugin = $this;
+            // Ground
+            $selection = $styler->getSelection(99998);
+            $plotBeginPos->y = 1;
+            $selection->setPosition(1, $plotBeginPos);
+            $selection->setPosition(2, new Vector3($plotBeginPos->x + $plotSize, $plotLevel->groundHeight, $plotBeginPos->z + $plotSize));
+            $cuboid = Cuboid::fromSelection($selection);
+            //$cuboid = $cuboid->async();
+            $cuboid->set($plotBeginPos->level, $plotFillBlock, function (float $time, int $changed) use ($plugin) : void {
+                $plugin->getLogger()->debug('Set ' . number_format($changed) . ' blocks in ' . number_format($time, 10) . 's');
+            });
+            $styler->removeSelection(99998);
+            // Bottom of world
+            $selection = $styler->getSelection(99998);
+            $plotBeginPos->y = 0;
```

Comment:
Should fill change the bedrock layer too?

Reply:
It currently leaves the bedrock layer. I think replacing it isn't what most people want when using /p fill (e.g. with air).